ginga/colors.py

```diff
@@ -755,27 +757,95 @@ def recalc_color_list():
 
 
 def lookup_color(name, format='tuple'):
-    color = None
-    if name.startswith('#'):
-        # hex notation
-        name = name[1:]
-        color = (int(name[:2], 16) / 255.0,
-                 int(name[2:4], 16) / 255.0,
-                 int(name[4:6], 16) / 255.0)
-    else:
-        # must be a name
-        try:
-            color = color_dict[name]
-        except KeyError:
-            raise KeyError("%s color does not exist in color_dict" % name)
+    """Find RGB or hex values for a supported color.
+
+    name : str
+        Color name (e.g., ``'red'``) or hash (e.g., ``'#ff0000'``).
+        Color name is case-sensitive.
+
+    format : {'tuple', 'hash', 'name'}
+        Desired output to be an RGB tuple, hash, or name.
+
+    Returns
+    -------
+    color : tuple or str
+        Color value as specified by ``format``.
+
+    Raises
+    ------
+    KeyError
+        Color name is not supported.
+
+    ValueError
+        Invalid format.
+
+    """
+    supported_formats = ('tuple', 'hash', 'name')
+    if format not in supported_formats:
+        raise ValueError(f'format needs to be one of {supported_formats}')
+
+    if name.startswith('#'):  # hex notation
+        if format == 'hash':
+            return name  # no-op
+
+        name = name[1:]  # Strip leading #
+        rgb = (int(name[:2], 16) / 255.0,
+               int(name[2:4], 16) / 255.0,
+               int(name[4:6], 16) / 255.0)
+        if format == 'tuple':
+            return rgb
+
+        color = _reverse_lookup(rgb)
+        if color is None:
+            raise KeyError(f'No color name for resolved RGB {rgb}')
 
-    if format == 'tuple':
         return color
-    elif format == 'hash':
-        return "#%02x%02x%02x" % (
-            int(color[0] * 255), int(color[1] * 255), int(color[2] * 255))
-    else:
-        raise ValueError("format needs to be 'tuple' or 'hash'")
+
+    # must be a name
+
+    elif name not in color_dict:
+        raise KeyError(f'{name} does not exist in color_dict')
+
+    if format == 'name':
+        return name  # no-op
+
+    rgb = color_dict[name]
+    if format == 'tuple':
+        return rgb
+
+    return f'#{int(rgb[0] * 255):02x}{int(rgb[1] * 255):02x}{int(rgb[2] * 255):02x}'
+
+
```

Comment:
@pllim, do you think the reverse color lookup is necessary?  I feel like since we can specify hex triplets to both ginga and matplotlib there is no real reason for it.  The color list in `colors` is obviously a small subset of the 24M possible colors.  It is not like we need to find the closest, non-perfect match when you can specify an exact triplet.  If it is easier to read the code with names, can do something like
```python
green, purple = '#1b9e77', '#7570b3'
```
inside a specific module (e.g. `ginga.util.plots`) and then use the variables in the code.

I don't see much of a use case for it, but if you really like it I'm ok to leave it in...

Reply:
Heh heh. I guess not. It was a nice programming exercise regardless... 😬